tasks/calamari_setup.py

```diff
@@ -179,90 +218,117 @@ def calamari_install(config, cal_svr):
+        delete_iceball = True
+    else:
+        prefix = os.path.join(iceball_loc,
+            'ICE-{0}-{1}'.format(ice_version, ice_distro))
+        for name in [prefix + ext for ext in ('.iso', '.tar.gz')]:
+            if os.path.exists(name):
+                iceball_file = name
+                break
+        else:
+            raise RuntimeError(
+                'Can''t find {0} at {1}'.format(prefix, iceball_loc)
             )
-            subprocess.check_call([exec_ice, '-I', ice_version,
-                                   '-b', version, '-o', ice_distro])
-            delete_iceball = True
-        except subprocess.CalledProcessError:
-            raise RuntimeError('%s failed for %s distro' %
-                               (exec_ice, ice_distro))
-        subprocess.check_call('rm -rf teuth-virtenv'.split(),
-                              cwd=ice_tool_loc)
 
-    gz_file = 'ICE-{0}-{1}.tar.gz'.format(ice_version, ice_distro)
-    lgz_file = os.path.join(iceball_loc, gz_file)
-    cal_svr.put_file(lgz_file, os.path.join('/tmp/', gz_file))
-    ret = cal_svr.run(args=['gunzip', run.Raw('<'), "/tmp/%s" % gz_file,
-                      run.Raw('|'), 'tar', 'xvf', run.Raw('-')])
+    remote_iceball_file = os.path.join('/tmp', os.path.split(iceball_file)[1])
+    cal_svr.put_file(iceball_file, remote_iceball_file)
+    if iceball_file.endswith('.tar.gz'):   # XXX specify tar/iso in config?
+        icetype = 'tarball'
+    elif iceball_file.endswith('.iso'):
+        icetype = 'iso'
+    else:
+        raise RuntimeError('Can''t handle iceball {0}'.format(iceball_file))
+
+    if icetype == 'tarball':
+        ret = cal_svr.run(args=['gunzip', run.Raw('<'), remote_iceball_file,
+                          run.Raw('|'), 'tar', 'xvf', run.Raw('-')])
+        if ret.exitstatus:
+            raise RuntimeError('remote iceball untar failed')
+    elif icetype == 'iso':
+        mountpoint = '/mnt/'   # XXX create?
+        ret = cal_svr.run(
+            args=['sudo', 'mount', '-r', remote_iceball_file, mountpoint,]
+        )
+
+    # install ice_setup package
+    args = {
+        'deb':'sudo dpkg -i /mnt/ice-setup*deb',
+        'rpm':'sudo rpm -i /mnt/ice_setup*rpm'
+    }.get(cal_svr.system_type, None)
+    if not args:
+        raise RuntimeError('{0}: unknown system type'.format(cal_svr))
+    import pdb; pdb.set_trace()
+    ret = cal_svr.run(args=args)
     if ret.exitstatus:
-        raise RuntimeError('remote tar failed')
+        raise RuntimeError('ice_setup package install failed')
+
+    # Run ice_setup
     icesetdata = 'yes\n\n%s\nhttp\n' % client_id
     ice_in = StringIO(icesetdata)
-    ice_setup_io = StringIO()
-    ret = cal_svr.run(args=['sudo', 'python', 'ice_setup.py'], stdin=ice_in,
-                      stdout=ice_setup_io)
-    log.debug(ice_setup_io.getvalue())
-    # Run Calamari-ceph connect.
+    ice_out = StringIO()
+    args = 'sudo ice_setup'
+    if icetype == 'iso':
+        args += ' -d /mnt'
```

Comment:
Do you foresee any need to handle RPMs that aren't signed?  By default, ice_setup will enable GPGCHECK in the .repo files.  if you pass `--nogpg`, it will disable. Useful for testing out beta .iso's that don't have signed RPMs yet.  But I'm not sure that's a usecase you need to worry about here.

Also, there is no capability for that in the .deb packaging yet.